src/django_components/util/django_monkeypatch.py

```diff
@@ -99,9 +101,32 @@ def _template_render(self: Template, context: Context, *args: Any, **kwargs: Any
             if context.template is None:
                 with context.bind_template(self):
                     context.template_name = self.name
-                    return self._render(context, *args, **kwargs)
+                    result = self._render(context, *args, **kwargs)
             else:
-                return self._render(context, *args, **kwargs)
+                result = self._render(context, *args, **kwargs)
+
+        # If the key is present, that means this Template is rendered as part of `Component.render()`
+        # or `{% component %}`. In that case the parent component will take care of rendering the
+        # dependencies, so we don't need to do that here.
+        if _COMPONENT_CONTEXT_KEY in context:
+            return result
+
+        # Allow users to configure the `deps_strategy` kwarg of `render_dependencies()`, even if
+        # they render a Template directly with `Template.render()` or Django's `django.shortcuts.render()`.
+        #
+        # Example:
+        # ```
+        # result = render_dependencies(
+        #     result,
+        #     Context({ "DJC_DEPS_STRATEGY": "fragment" }),
+        # )
+        # ```
+        if _STRATEGY_CONTEXT_KEY in context and context[_STRATEGY_CONTEXT_KEY] is not None:
```

Comment:
This is the new logic added inside `Template.render()`

After removing the middleware, there was a question of how to configure which rendering strategy should be used when one renders templates with `Template.render()` or `django.shortcuts.render()`.

What I went with is that you can set the strategy by setting a context variable `DJC_DEPS_STRATEGY`.

If `DJC_DEPS_STRATEGY` is omitted, then the default strategy ("document") is used, in which case the behaviour is the same as it was when using the middleware.